client/python/modeldb/__main__.py

```diff
@@ -0,0 +1,182 @@
+import os
+import json
+
+
+def to_bool(value):
+    valid = {'true': True, 't': True, '1': True,
+             'false': False, 'f': False, '0': False,
+             }
+
+    if isinstance(value, bool):
+        return value
+
+    if not isinstance(value, basestring):
+        raise ValueError('invalid literal for boolean. Not a string.')
+
+    lower_value = value.lower()
+    if lower_value in valid:
+        return valid[lower_value]
+    else:
+        raise ValueError('invalid literal for boolean: "%s"' % value)
+
+
+if __name__ == '__main__':
+    execution_dir = os.getcwd()
+    file_dir = os.path.dirname(__file__)
+    filename = file_dir + '/syncer.json'
+    config = json.load(file(filename, 'r'))
+    new_config = json.load(file(filename, 'r'))
+
+    prompt = '> '
+
+    raw_input(
+        'This utility helps you create a modeldb configuration file.\n'
+        'The file is used by the modeldb python and/or scala clients to '
+        'connect to the modeldb server.\n\n'
+        'Press <ENTER> to use defaults')
+
+    print('\n== Output Filename ==')
+    prompt = 'output filename (default=custom_syncer.json): '
+    output_filename = str(raw_input(prompt))
+    if output_filename == '':
+        output_filename = 'custom_syncer.json'
+
+    print('\n== Thrift ==')
+
+    prompt = 'host (default=%s): ' % config['thrift']['host']
+    thrift_host = raw_input(prompt)
+    if thrift_host == '':
+        thrift_host = config['thrift']['host']
+    new_config['thrift']['host'] = thrift_host
+
+    prompt = 'port (default=%s): ' % (config['thrift']['port'])
+    thrift_port = raw_input(prompt)
+    if thrift_port == '':
+        thrift_port = config['thrift']['port']
+    thrift_port = int(thrift_port)
+    new_config['thrift']['port'] = thrift_port
+
+    print('\n== Syncing Strategy =='
+          'eager/lazy')
+
+    prompt = 'syncing strategy (default=%s): ' % config['syncingStrategy']
+    syncing_strategy = raw_input(prompt)
+    if syncing_strategy == '':
+        syncing_strategy = config['syncingStrategy']
+    new_config['syncingStrategy'] = syncing_strategy
+
+    print('\n== Project ==')
+    prompt = 'name (default=%s): ' % (config['project']['name'])
+    project_name = raw_input(prompt)
+    if project_name == '':
+        project_name = config['project']['name']
+    new_config['project']['name'] = project_name
+
+    prompt = 'author (default=%s): ' % (config['project']['author'])
+    project_author = raw_input(prompt)
+    if project_author == '':
+        project_author = config['project']['author']
+    new_config['project']['author'] = project_author
+
+    prompt = 'description (default=%s): ' % (config['project']['description'])
```

Comment:
If I replaced the slightly nicer formatting, it'd be possible to use the keys already supplied in syncer.json. This wouldn't look quite as good, but would reduce the number of lines of code. Would that be useful?